### PR TITLE
Update UG structure

### LINKED ISSUE #77: [PE-D] No example of date format
When I execute "addcheese q/1 t/brie d/0" which is an incorrect date format. It only displays some date format examples with no actual example input. Some users may not know the difference between "MM" and "MMM".

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: yeoutzer/ped#14

### LINKED ISSUE #80: [PE-D] User Guide notes on command format
In the User Guide under "notes about command format", there is a mention where items in square brackets are optional. However, the example used was "listcheeses" does not even have any optional inputs available mention later in the User Guide. This may cause confusion for some users reading the guide. May be good to use another command as an example.

Under notes about command format

listcheeses in User Guide

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: yeoutzer/ped#17

### LINKED ISSUE #81: [PE-D] findcustomer in User Guide
For the "findcustomer" in the User Guide, it is not that clear on how the application will find customers by. For example, would "findcustomer m/ber p/9" display any customer as long as they have "ber" in their name or "9" in their phone number or would it only display customers that have "ber" in their name and "9" in their phone number. Meaning, must all the fields match to display the contact? Or as long as 1 field matches, it will display the contact.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: yeoutzer/ped#18

### LINKED ISSUE #84: [PE-D] Command descriptions not well organised
The command descriptions are not as well organised. Currently, they are all grouped by command type e.g. all the edit commands together.

However, a user is unlikely to look at the user guide like this. Perhaps you could group them according to the specific feature e.g. order commands together, customer commands together.

It was difficult to navigate as I constantly had to jump to the different sections.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: jasaaanlim/ped#2

### LINKED ISSUE #102: [PE-D] UG - find command format mistake
All the find command formats specify that **all** parameters are optional when atleast 1 parameter must be specified.

Example the UG shows the format for `findcustomer` as 

`findcustomer [n/NAME_KEYWORDS...] [p/PHONE_KEYWORDS...] [e/EMAIL_KEYWORDS...] [a/ADDRESS_KEYWORDS...]`

This means that even if no parameters are passed, the command should work, which is not the case. Same issue with `findcheese` and `findorder`.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: noelmathewisaac/ped#8

### LINKED ISSUE #103: [PE-D] UG - Notes about command format
In the `Notes about the command format` section, add a point mentioning that items with `...` can be used multiple times. Example: `[t/tags...]` means 0 or more tags. 

Otherwise, users may get confused on seeing the syntax.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: noelmathewisaac/ped#9

### LINKED ISSUE #105: [PE-D] Wrong Date being added with addorder command
When date is higher than the number of days in the month but less than or equal to 31, an order with the last day of the month is added. (eg Nov 31-> Nov 30). An error message should be displayed instead.

Example:

`addorder t/Parmesan q/2 p/123456789 d/2021-02-31` will add an order with date as 28 Feb 2021 instead of displaying an error message.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: noelmathewisaac/ped#11

### LINKED ISSUE #108: [PE-D] addcheese showing manufacture date as default date
user input:
addcheese t/cheese test 3 q/1 

result:

if no date is specified, manufacture date is assumed to be today's date. is that the expected behavior? 

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: wei-yutong/ped#3

### PR DESCRIPTION
**Pull request by laurenlhy**
Closes #77
Closes #81  
Closes #84 
Closes #102  
Closes #108

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -10,11 +10,8 @@ The app is optimized for use via a Command Line Interface (CLI) while still havi
 benefits of a Graphical User Interface (GUI). If you can type fast, CHIM can get your management tasks done
 faster than traditional GUI apps.
 
-* Table of Content
-  - [Quick start](#quick-start)
```
Comment (daniellau88) [inline docs/UserGuide.md:14]: Any reason to remove?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -44,179 +44,246 @@ faster than traditional GUI apps.
...
+#### Finding customers: `findcustomer`
+
+Searches for a customer in CHIM.
 
-Format: `editcustomer INDEX [n/CUSTOMER_NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/tags...]`
+Format: `findcustomer [n/NAME_KEYWORDS…] [p/PHONE_KEYWORDS…] [e/EMAIL_KEYWORDS…] [a/ADDRESS_KEYWORDS…] `
+* At least one of the optional fields must be provided to find a customer.
+* Search is case-insensitive, e.g. Betty will match betty.
+* Search will search by given keywords as prefix, e.g. Bet will match Betty.
+* Search will find any customers which match all of the given fields,
```
Comment (daniellau88) [inline docs/UserGuide.md]: maybe can bold the word all

File: docs/UserGuide.md [hunk 3]
```diff
@@ -44,179 +44,246 @@ faster than traditional GUI apps.
...
+#### Editing a cheese: `editcheese`
 
-The cheese must be unassigned.
+Edits an existing unassigned cheese in CHIM.
+
+Format: `editcheese INDEX [t/CHEESE_TYPE] [d/MANUFACTURE_DATE] [d/EXPIRY_DATE]`
+* At least one of the optional fields must be provided to edit a cheese.
+* The specified `INDEX` must be a positive integer.
+* All dates must be given in these formats: `DD/MM/YYYY` or `YYYY-MM-DD` or `MMM DD YYYY`.
+* Expiry Date specified must occur after the Manufacture Date.
```
Comment (daniellau88) [inline docs/UserGuide.md]: Change Expiry Date and Manufacture Date to caps and ``

File: docs/UserGuide.md (hunk 1 above)
Comment (laurenlhy) [reply docs/UserGuide.md:14]: I wanted to use the same autogenerated TOC as DG

File: docs/UserGuide.md [hunk 4]
```diff
@@ -44,179 +44,246 @@ faster than traditional GUI apps.
...
+* `addcheese t/Gouda q/3`
+* `addcheese t/Parmesan q/2 d/2021-03-12 e/2021-05-12`
 
-Deletes a specified cheese from the inventory.
+#### Editing a cheese: `editcheese`
 
-The cheese must be unassigned.
+Edits an existing unassigned cheese in CHIM.
+
+Format: `editcheese INDEX [t/CHEESE_TYPE] [d/MANUFACTURE_DATE] [d/EXPIRY_DATE]`
```
Comment (AhQuanz) [inline docs/UserGuide.md]: i think the d/EXPIRY_DATE needs to be change to e?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -44,179 +44,246 @@ faster than traditional GUI apps.
...
 
-### Editing a cheese: `editcheese`
+#### Deleting a customer: `deletecustomer`
 
-Edits an existing unassigned cheese in CHIM.
+Deletes a customer from the list of customers, using their phone number as an identifier.
 
-Format: `editcheese INDEX [t/CHEESE_TYPE] [d/MANUFACTURE_DATE] [d/EXPIRY_DATE]`
-* The specified `INDEX` must be a positive integer.
+If the customer has orders, their orders will be deleted by CHIM.
```
Comment (AhQuanz) [inline docs/UserGuide.md]: WIll be good to mention that if there are some orders completed , the cheeses assigned will be deleted as well.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -44,179 +44,246 @@ faster than traditional GUI apps.
...
-Format: `listcustomers`
+Edits an existing uncompleted order in CHIM.
 
-### Marking an order as complete: `done`
+Format: `editorder INDEX [t/CHEESE_TYPE] [q/QUANTITY] [p/PHONE_NUMBER] [d/ORDER_DATE]`
+* At least one of the optional fields must be provided to edit an order.
+* The specified `INDEX` and `QUANTITY` must be a positive integer.
+* The specified `CHEESE_TYPE` need not belong to an existing cheese type in CHIM.
+* The specified `PHONE_NUMBER` must belong to an existing user.
+* The specified `ORDER_DATE` must be given in these formats: `DD/MM/YYYY` or `YYYY-MM-DD` or `MMM DD YYYY`.
```
Comment (AhQuanz) [inline docs/UserGuide.md:216]: i think there is a missing note of 'ORDER_DATE' must be any date .....

**Review by AhQuanz — APPROVED** [review]
LGTM except for that comment.

File: docs/UserGuide.md [hunk 7]
```diff
@@ -44,179 +44,246 @@ faster than traditional GUI apps.
...
 
 
+### Saving the data
+
+CHIM data are saved in the hard disk automatically after any command that changes the data. There is no need to save manually.
+
+
+### Editing the data file
+
+CHIM data are saved as a JSON file `[JAR file location]/data/chim.json`. Advanced users are welcome to update data directly by editing that data file.
```
Comment (weixue123) [inline docs/UserGuide.md:282]: May I confirm if this is the case for our project? I tried running the JAR file but couldn't find the data folder anywhere. The storage works perfectly though.

File: docs/UserGuide.md (hunk 6 above)
Comment (weixue123) [reply docs/UserGuide.md:216]: I think ORDER_DATE can only be today or earlier

File: docs/UserGuide.md (hunk 7 above)
Comment (laurenlhy) [reply docs/UserGuide.md:282]: The data folder appears for me.

Comment (AhQuanz) [reply docs/UserGuide.md:282]: its a hidden folder .data/chim,json

Comment (weixue123) [reply docs/UserGuide.md:282]: okay got it thanks

**Review by weixue123 — APPROVED** [review]

**Pull request merged**